pkg/decode/decode.go

```diff
@@ -806,6 +806,14 @@ func (d *D) FieldRangeFn(name string, firstBit int64, nBits int64, fn func() *Va
 	return v
 }
 
+func (d *D) AssertPosBytes(nBytes int64) {
```

Comment:
Maybe make more general and use bit position? `AssertPos(pos int64)` 

There is a help type to format bit positions as `#bytes.#bits-into-byte`, ex bit pos 17 is "2.1". Something like:
```go
panic(DecoderError{Reason: fmt.Sprintf("expected position %s", mathex.Bits(pos).StringByteBits(10)), Pos: d.Pos()})
```